removes addLocal flag for Rancher version upgrade

ddl-oscarluu opened the pull request:
To upgrade to a newer Rancher version >= 2.5.x, the addLocal flag must be removed as Rancher will not start with it in `false`. This PR is to remove the flag, and add `restrictedAdmin` in order for us to upgrade Rancher to latest (2.5.6).
https://rancher.com/docs/rancher/v2.x/en/admin-settings/rbac/global-permissions/#upgrading-from-rancher-with-a-hidden-local-cluster

The `restrictedAdmin` flag is now used to continue restricting access to the local cluster.

sonnysideup reviewed (changes requested):
This isn't backwards compatible; see https://github.com/dominodatalab/ranchhand/blob/c07e25e19839142a21c0ef26fb0135380a864fa9/ansible/roles/rancher/vars/main.yml#L4. Either the main version needs to change and we should add something to the README that indicates Rancher version >= 2.5, or these flags need to change based on the provided rancher version variables.

/cc @steved

ddl-oscarluu commented:
> This isn't backwards compatible; see https://github.com/dominodatalab/ranchhand/blob/c07e25e19839142a21c0ef26fb0135380a864fa9/ansible/roles/rancher/vars/main.yml#L4. Either the main version needs to change and we should add something to the README that indicates Rancher version >= 2.5, or these flags need to change based on the provided rancher version variables.
> 
> 
> 
> /cc @steved 

Sounds good, I'll address and I can discuss with @steved on best paths forward. Thanks!

ddl-oscarluu commented:
@sonnysideup I changed the main version to 2.5.6 and added a quip to the README to indicate the version requirement. Let me know if there's anything else I need to address. Thanks!

sonnysideup commented:
@steved do you know if we're still going to run this in such a manner that there will be a flag incompatibility? I'm too far removed to have a reliable notion at this point.

steved commented:
We don't currently re-run this on existing installs, so this should be fine. In fact, we no longer use this anywhere but AWS and can probably consider bringing it back in-house.
